src/io/msh.rs

```diff
@@ -0,0 +1,285 @@
+        .take()
+        .ok_or(eyre!("MSH file does not contain elements"))?;
+
+    let mut vertices = Vec::new();
+    let mut connectivity = Vec::new();
+
+    // Ensure that at least one element block matches the target mesh connectivity
+    if !msh_elements
+        .element_blocks
+        .iter()
+        .any(|block| element_block_matches_connectivity::<C, _>(block))
+    {
+        return Err(eyre!(
+            "MSH file does not contain an element block of the requested type (type {:?} of with reference/entity dim {})",
+            C::msh_element_type(),
+            C::reference_dim()
+        ));
+    }
+
+    // Collect all mesh vertices
+    for node_block in &msh_nodes.node_blocks {
+        let block_vertices = vertices_from_node_block(node_block)?;
+        vertices.extend(block_vertices);
+    }
+
+    if vertices.len()
+        != msh_nodes
+            .num_nodes
+            .to_usize()
+            .expect("failed to convert num_nodes to usize")
+    {
+        return Err(eyre!(
+            "only {} vertices were read but msh file claims to contain {} nodes",
+            vertices.len(),
+            msh_nodes.num_nodes
+        ));
+    }
+
+    // Collect all connectivity matching the target connectivity
+    for element_block in &msh_elements.element_blocks {
+        let block_connectivity = connectivity_from_element_block(element_block)?;
+        connectivity.extend(block_connectivity);
+    }
+
+    Ok(Mesh::from_vertices_and_connectivity(vertices, connectivity))
+}
+
+/// Tries to convert a `mshio::NodeBlock` to a `Vec<OPoint<T, D>>`.
+fn vertices_from_node_block<T, D, F, I>(node_block: &mshio::NodeBlock<u64, I, F>) -> eyre::Result<Vec<OPoint<T, D>>>
+where
+    T: RealField,
+    D: DimName,
+    F: mshio::MshFloatT,
+    I: mshio::MshIntT,
+    DefaultAllocator: Allocator<T, D>,
+{
+    // Ensure that node tags are consecutive
+    if node_block.node_tags.is_some() {
+        return Err(eyre!(
+            "node block tags are not consecutive in msh file (sparse tags are not supported)"
+        ));
+    }
+
+    // Note: The MSH `node_block`'s `entity_dim` does not seem to correspond to the geometrical
+    //  dimension of the points. Rather it seems to correspond to the dimension of the "physical"
+    //  object represented by the node block.
+    //  When creating primitives in Gmsh for example, the nodes of a triangulation of a sphere are
+    //  divided into node_blocks representing its equator, surface and volume and all of them are
+    //  referenced by the volumetric elements.
+    //  In addition, all node blocks have to be read in order for the global `node_tag` indexing to
+    //  be consistent work.
+    /*
+    if node_block
+        .entity_dim
+        .to_usize()
+        .ok_or_else(|| eyre!("error converting node block entity dimension to usize"))?
+        != D::dim()
+    {
+        return Err(eyre!("node block entity does not have the right dimension for this mesh"));
+    }
+    */
```

Comment:
Is this still relevant? Should it be left in but commented? It's not clear to me why it's commented out, which probably means it should either be removed or fixed? (Or replaced with a TODO or something)